### ISSUE TITLE
CHAT_ADMIN_REQUIRED

### ISSUE THREAD

**Issue by StegaDP** [pyrogram/pyrogram#1405]
### Checklist

- [X] I am sure the error is coming from Pyrogram's code and not elsewhere
- [X] I have searched in the issue tracker for similar bug reports, including closed ones
- [X] I ran `pip3 install -U https://github.com/pyrogram/pyrogram/archive/master.zip` and reproduced the issue using the latest development version

### Description

pyrogram.errors.exceptions.bad_request_400.ChatAdminRequired: Telegram says: [400 CHAT_ADMIN_REQUIRED] - The method requires chat admin privileges (caused by "stats.GetMegagroupStats")
app.invoke is throwing this. i'm chat owner(means i'm admin as well lol)

### Steps to reproduce

async def main():
    async with Client("my_account", api_id, api_hash) as app:
        r = await app.invoke(GetMegagroupStats(channel=await app.resolve_peer(super_group_id)))
        print(r)

if __name__ == '__main__':
    asyncio.run(main())

**Comment by Likheet** [comment]
This error is likely occurring because the GetMegagroupStats method requires the client to be an administrator in the supergroup. Please assign me this task and I will do my best to fix this issue

**Comment by StegaDP** [comment]
not really, i'm a group owner, as well as being admin. more likely to be some bug on telegram side idk


![image](https://github.com/pyrogram/pyrogram/assets/102144407/b3fbda24-726b-4502-ab81-ebc6294d7604)

**Comment by Likheet** [comment]
Did u try GetChatMembers instead of GetMegagroupStats ? But it may not be as detailed

**Comment by StegaDP** [comment]
well, that's what i needed first. i will try getchatmembers but it doesn't seem to be thing i'm looking for

**Comment by Likheet** [comment]
My logic is, there may be something which is not allowing u to be set as admin. Try getting info on yourself using get_chat_member. I think something similar to this:

`from pyrogram import Client

app = Client("my_account")

@app.on_message()
async def check_admin(client, message):
    chat_member = await client.get_chat_member(message.chat.id, message.from_user.id)
    if chat_member.status == "administrator" or chat_member.status == "creator":
        await message.reply_text("Admin")
    else:
        await message.reply_text("Not Admin")

app.run()`
